### PR TITLE
Adding Options to MarginBalance due to missing fields in response

### PR DESCRIPTION
**Pull request by Verticaleap**
Adding Options around the fields I've found could be absent from responses from the Schwab Trader API. If there is an easier way to get around missing fields, I'm all ears!

Great crate!

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -12,3 +12,6 @@ Cargo.lock
 
 # MSVC Windows builds of rustc generate these, which store debugging information
 *.pdb
+
+# Vim sawp files
```
Comment (z-Wind) [inline .gitignore]: Do you mean Vim swap files ???

File: src/model/trader/accounts.rs [hunk 2]
```diff
@@ -66,42 +66,61 @@ pub struct MarginInitialBalance {
...
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub short_margin_value: Option<f64>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub sma: Option<f64>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub is_in_call: Option<bool>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub stock_buying_power: Option<f64>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub option_buying_power: Option<f64>,
```
Comment (z-Wind) [inline src/model/trader/accounts.rs:105]: You can use `#[skip_serializing_none]` in `serde_with`
https://docs.rs/serde_with/latest/serde_with/#skip_serializing_none

**Comment by z-Wind** [comment]
Could you provide a test case for [Accounts_real.json](https://github.com/z-Wind/schwab_api/blob/main/tests/model/Trader/Accounts_real.json)? If possible, please remember to remove any sensitive information.

File: .gitignore (hunk 1 above)
Comment (Verticaleap) [reply .gitignore]: I absolutely did

File: src/model/trader/accounts.rs (hunk 2 above)
Comment (Verticaleap) [reply src/model/trader/accounts.rs:105]: Haven't seen seen this crate before, I like it.

**Comment by z-Wind** [comment]
Thanks for contribution. Merging this PR for now.

**Pull request merged**